diff --git a/homeassistant/components/homekit/__init__.py b/homeassistant/components/homekit/__init__.py
index 0626071e..22608e22 100644
--- a/homeassistant/components/homekit/__init__.py
+++ b/homeassistant/components/homekit/__init__.py
@@ -88,6 +88,7 @@ from . import (  # noqa: F401
     type_cameras,
     type_covers,
     type_fans,
+    type_heatercooler,
     type_humidifiers,
     type_lights,
     type_locks,
diff --git a/homeassistant/components/homekit/accessories.py b/homeassistant/components/homekit/accessories.py
index 186be4ce..96c65058 100644
--- a/homeassistant/components/homekit/accessories.py
+++ b/homeassistant/components/homekit/accessories.py
@@ -87,10 +87,12 @@ from .const import (
     TYPE_AIR_PURIFIER,
     TYPE_FAN,
     TYPE_FAUCET,
+    TYPE_HEATER_COOLER,
     TYPE_OUTLET,
     TYPE_SHOWER,
     TYPE_SPRINKLER,
     TYPE_SWITCH,
+    TYPE_THERMOSTAT,
     TYPE_VALVE,
 )
 from .iidmanager import AccessoryIIDStorage
@@ -117,6 +119,10 @@ FAN_TYPES = {
     TYPE_AIR_PURIFIER: "AirPurifier",
     TYPE_FAN: "Fan",
 }
+CLIMATE_TYPES = {
+    TYPE_HEATER_COOLER: "HeaterCooler",
+    TYPE_THERMOSTAT: "Thermostat",
+}
 TYPES: Registry[str, type[HomeAccessory]] = Registry()
 
 RELOAD_ON_CHANGE_ATTRS = (
@@ -151,7 +157,10 @@ def get_accessory(  # noqa: C901
         a_type = "BinarySensor"
 
     elif state.domain == "climate":
-        a_type = "Thermostat"
+        if climate_type := config.get(CONF_TYPE):
+            a_type = CLIMATE_TYPES[climate_type]
+        else:
+            a_type = "Thermostat"
 
     elif state.domain == "cover":
         device_class = state.attributes.get(ATTR_DEVICE_CLASS)
diff --git a/homeassistant/components/homekit/const.py b/homeassistant/components/homekit/const.py
index 5d5a8efc..5c1ed79f 100644
--- a/homeassistant/components/homekit/const.py
+++ b/homeassistant/components/homekit/const.py
@@ -128,6 +128,10 @@ TYPE_VALVE = "valve"
 TYPE_FAN = "fan"
 TYPE_AIR_PURIFIER = "air_purifier"
 
+# #### Climate Types ####
+TYPE_HEATER_COOLER = "heater_cooler"
+TYPE_THERMOSTAT = "thermostat"
+
 # #### Categories ####
 CATEGORY_RECEIVER = 34
 
@@ -145,6 +149,7 @@ SERV_DOORBELL = "Doorbell"
 SERV_FANV2 = "Fanv2"
 SERV_FILTER_MAINTENANCE = "FilterMaintenance"
 SERV_GARAGE_DOOR_OPENER = "GarageDoorOpener"
+SERV_HEATER_COOLER = "HeaterCooler"
 SERV_HUMIDIFIER_DEHUMIDIFIER = "HumidifierDehumidifier"
 SERV_HUMIDITY_SENSOR = "HumiditySensor"
 SERV_INPUT_SOURCE = "InputSource"
@@ -193,6 +198,7 @@ CHAR_CURRENT_AMBIENT_LIGHT_LEVEL = "CurrentAmbientLightLevel"
 CHAR_CURRENT_AIR_PURIFIER_STATE = "CurrentAirPurifierState"
 CHAR_CURRENT_DOOR_STATE = "CurrentDoorState"
 CHAR_CURRENT_FAN_STATE = "CurrentFanState"
+CHAR_CURRENT_HEATER_COOLER_STATE = "CurrentHeaterCoolerState"
 CHAR_CURRENT_HEATING_COOLING = "CurrentHeatingCoolingState"
 CHAR_CURRENT_HUMIDIFIER_DEHUMIDIFIER = "CurrentHumidifierDehumidifierState"
 CHAR_CURRENT_POSITION = "CurrentPosition"
@@ -245,6 +251,7 @@ CHAR_STREAMING_STRATUS = "StreamingStatus"
 CHAR_SWING_MODE = "SwingMode"
 CHAR_TARGET_AIR_PURIFIER_STATE = "TargetAirPurifierState"
 CHAR_TARGET_DOOR_STATE = "TargetDoorState"
+CHAR_TARGET_HEATER_COOLER_STATE = "TargetHeaterCoolerState"
 CHAR_TARGET_HEATING_COOLING = "TargetHeatingCoolingState"
 CHAR_TARGET_POSITION = "TargetPosition"
 CHAR_TARGET_FAN_STATE = "TargetFanState"
diff --git a/homeassistant/components/homekit/util.py b/homeassistant/components/homekit/util.py
index 97269181..07613a72 100644
--- a/homeassistant/components/homekit/util.py
+++ b/homeassistant/components/homekit/util.py
@@ -106,10 +106,12 @@ from .const import (
     TYPE_AIR_PURIFIER,
     TYPE_FAN,
     TYPE_FAUCET,
+    TYPE_HEATER_COOLER,
     TYPE_OUTLET,
     TYPE_SHOWER,
     TYPE_SPRINKLER,
     TYPE_SWITCH,
+    TYPE_THERMOSTAT,
     TYPE_VALVE,
     VIDEO_CODEC_COPY,
     VIDEO_CODEC_H264_OMX,
@@ -217,6 +219,20 @@ FAN_SCHEMA = BASIC_INFO_SCHEMA.extend(
     }
 )
 
+CLIMATE_SCHEMA = BASIC_INFO_SCHEMA.extend(
+    {
+        vol.Optional(CONF_TYPE): vol.All(
+            cv.string,
+            vol.In(
+                (
+                    TYPE_HEATER_COOLER,
+                    TYPE_THERMOSTAT,
+                )
+            ),
+        ),
+    }
+)
+
 COVER_SCHEMA = BASIC_INFO_SCHEMA.extend(
     {
         vol.Optional(CONF_LINKED_OBSTRUCTION_SENSOR): cv.entity_domain(
@@ -360,6 +376,9 @@ def validate_entity_config(values: dict) -> dict[str, dict]:
         elif domain == "humidifier":
             config = HUMIDIFIER_SCHEMA(config)
 
+        elif domain == "climate":
+            config = CLIMATE_SCHEMA(config)
+
         elif domain == "cover":
             config = COVER_SCHEMA(config)
 
